### PR TITLE
Consolidating beacon configuration mapping code

### LINKED ISSUE #60241: [BUG] inotify beacon fails with AttributeError: 'list' object has no attribute 'get'
**Description**
The inotify beacon fails after sending an event and spams the logfile with python stack traces.

**Setup**
```
beacons:
  enabled: true
  watch_important_files:
  - beacon_module: inotify
  - disable_during_state_run: true
  - files:
      /opt/newdir/do-not-touch:
        mask:
        - modify
```

**Steps to Reproduce the behavior**
1) salt minion1 beacons.add watch_important_files "[{'beacon_module': 'inotify'}, {'disable_during_state_run': true}, {'files': {'/opt/newdir/do-not-touch': {'mask': ['modify']}}}]"
2) modify /opt/newdir/do-not-touch

**Expected behavior**
the minion should not raise an error

**Screenshots**
```
2021-05-24 14:29:43,083 [salt.minion      :2947][CRITICAL][32837] The beacon errored: 
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/salt/minion.py", line 2945, in handle_beacons
    beacons = self.process_beacons(self.functions)
  File "/usr/lib/python3/dist-packages/salt/minion.py", line 512, in process_beacons
    return self.beacons.process(
  File "/usr/lib/python3/dist-packages/salt/beacons/__init__.py", line 107, in process
    self.beacons[close_str](b_config[mod])
  File "/usr/lib/python3/dist-packages/salt/loader.py", line 1235, in __call__
    return self.loader.run(run_func, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/salt/loader.py", line 2268, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/salt/loader.py", line 2283, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/salt/beacons/inotify.py", line 370, in close
    beacon_name = config.get("_beacon_name", "inotify")
AttributeError: 'list' object has no attribute 'get'
```

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
Salt Version:
          Salt: 3003
 
Dependency Versions:
          cffi: Not Installed
      cherrypy: Not Installed
      dateutil: 2.7.3
     docker-py: Not Installed
         gitdb: 2.0.6
     gitpython: 3.0.7
        Jinja2: 2.10.1
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 0.6.2
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: 3.6.1
        pygit2: Not Installed
        Python: 3.8.5 (default, Jan 27 2021, 15:41:15)
  python-gnupg: 0.4.5
        PyYAML: 5.3.1
         PyZMQ: 18.1.1
         smmap: 2.0.5
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.2
 
System Versions:
          dist: ubuntu 20.04 focal
        locale: utf-8
       machine: x86_64
       release: 5.4.0-73-generic
        system: Linux
       version: Ubuntu 20.04 focal
```
</details>

**Additional context**

### LINKED ISSUE #60755: inotify Beacon Throwing Exceptions on Minion
### Description of Issue
I am running an inotify beacon that looks for file changes and which calls an orchestration that sends me an email when a file is changed. Looking at the journal for salt-minion, I see multiple repeated exceptions.

I am also having some event loops where a beacon detects that the SSHd config changed, then calls an orchestration which highstates the minion and sends me an email. This beacon seems to get stuck in loops where it will detect a change, highstates, and does it again until the minion service is restarted. The beacon sls does set disable_during_state_run as True. This looping problem may not be related to this issue except that it causes these exceptions - the exception occurs when the file is changed.

### Setup: Minion beacons.conf
```#######################################################
# SSH Section
  SSHD_Files:
    - files:
        /etc/ssh/sshd_config:
          mask:
            - modify
    - disable_during_state_run: True
    - beacon_module: inotify
  SSHD_Service:
    - services:
        sshd:
          onchangeonly: True
    - beacon_module: service
    - disable_during_state_run: True
#######################################################
# Some sections in the beacon config do not apply to the minion it is running on, but it is in the minion's beacon config anyway.
  Svc_Agent_Files:
    - files:
        /etc/svc/agent.conf:
          mask:
            - modify
    - disable_during_state_run: True
    - beacon_module: inotify
  Svc_Agent_Service:
    - services:
        svc-agent:
          onchangeonly: True
    - beacon_module: service
    - disable_during_state_run: True
```

### Setup: salt-minion Journal
Hundreds of these exceptions can be called at once.
```
[CRITICAL] The beacon errored:
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/salt/minion.py", line 2945, in handle_beacons
    beacons = self.process_beacons(self.functions)
  File "/usr/lib/python3.6/site-packages/salt/minion.py", line 513, in process_beacons
    b_conf, self.opts["grains"]
  File "/usr/lib/python3.6/site-packages/salt/beacons/__init__.py", line 107, in process
    self.beacons[close_str](b_config[mod])
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 1235, in __call__
    return self.loader.run(run_func, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2268, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/contextvars/__init__.py", line 38, in run
    return callable(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2283, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/beacons/inotify.py", line 370, in close
    beacon_name = config.get("_beacon_name", "inotify")
AttributeError: 'list' object has no attribute 'get'
```

### Steps to Reproduce Issue
The exceptions and looping occur when the SSHd config file is modified. It also is likely to occur when other files are modified, though I have not confirmed this.

### Versions Report
```Salt Version:
          Salt: 3003.1
 
Dependency Versions:
          cffi: 1.11.5
      cherrypy: Not Installed
      dateutil: 2.6.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 2.10.1
       libgit2: 0.99.0
      M2Crypto: 0.35.2
          Mako: Not Installed
       msgpack: 0.5.6
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.14
      pycrypto: Not Installed
  pycryptodome: Not Installed
        pygit2: 1.1.1
        Python: 3.6.8 (default, Mar 19 2021, 05:13:41)
  python-gnupg: Not Installed
        PyYAML: 3.12
         PyZMQ: 19.0.0
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: centos 8 
        locale: UTF-8
       machine: x86_64
       release: 4.18.0-305.3.1.el8.x86_64
        system: Linux
       version: CentOS Stream 8```

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Move the commonly used code that converts a list to a dictionary into salt.utils.beacons.  Fixing inotify beacon close function to ensure the configuration is converted from the provided list format into a dictionary.

### What issues does this PR fix or reference?
Fixes: #60241 #60755

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: tests/pytests/unit/beacons/test_haproxy.py [hunk 1]
```diff
@@ -44,6 +48,7 @@ def test_threshold_reached():
     mock = MagicMock(return_value=46)
     with patch.dict(haproxy.__salt__, {"haproxy.get_sessions": mock}):
         ret = haproxy.beacon(config)
+        log.debug("=== ret %s ===", ret)
```
Comment (Ch3LL) [inline tests/pytests/unit/beacons/test_haproxy.py:51]: remove log statement.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
Just one small change request. Also, do you think there is a sufficient test coverage for all these beacons to ensure this change does not cause a regression?

**Comment by garethgreenaway** [comment]
@Ch3LL I'll add some tests for the functions in salt.utils.beacons.

**Review by Ch3LL — DISMISSED** [review]

**Comment by Ch3LL** [comment]
ping @waynew can you review here before a merge?

File: salt/utils/beacons.py [hunk 2]
```diff
@@ -14,3 +14,13 @@ def remove_hidden_options(config, whitelist):
...
     return config
+
+
+def list_to_dict(config):
+    """
+    Convert list based beacon configuration
+    into a dictionary.
+    """
+    _config = {}
+    list(map(_config.update, config))
```
Comment (waynew) [inline salt/utils/beacons.py:25]: just a question here - do we know if this is faster than

```
for thing in config:
    _config.update(thing)
```
?

Or
```
[_config.update(x) for x in config]
```
I don't know if it's important, but seeing how many different places this is used, that information would at least be interesting!

File: tests/pytests/unit/beacons/test_haproxy.py [hunk 3]
```diff
@@ -4,10 +4,14 @@
...
     HAProxy beacon test cases
 """
+import logging
+
 import pytest
 import salt.beacons.haproxy as haproxy
 from tests.support.mock import MagicMock, patch
 
+log = logging.getLogger(__name__)
+
```
Comment (waynew) [inline tests/pytests/unit/beacons/test_haproxy.py:14]: **question** do we need the import/create log here?

**Review by waynew — CHANGES_REQUESTED** [review]
Just one other question -- otherwise it looks... as good as reasonably possible :joy:

File: salt/utils/beacons.py (hunk 2 above)
Comment (garethgreenaway) [reply salt/utils/beacons.py:25]: @waynew FYI.

In [23]: %timeit list(map(_config.update, config))
392 ns ± 1.79 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [28]: %timeit [_config.update(x) for x in config]
292 ns ± 2.38 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [30]: %%timeit
    ...: for thing in config:
    ...:     _config.update(thing)
    ...: 
151 ns ± 0.366 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

File: tests/pytests/unit/beacons/test_haproxy.py (hunk 3 above)
Comment (garethgreenaway) [reply tests/pytests/unit/beacons/test_haproxy.py:14]: It's not being actively used here and can be removed if necessary.

**Review by Ch3LL — DISMISSED** [review]

**Comment by garethgreenaway** [comment]
re-run all

**Review by Ch3LL — DISMISSED** [review]

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**